homework/index.js

```diff
@@ -30,18 +36,63 @@
     return elem;
   }
 
+  
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+    fetchJSON(url)
+      .then(repositoryData => {
+        const select = createAndAppend('select', root);
+        createAndAppend('option', select, { text: 'Click here to choose a Repository' });
+      
+        repositoryData.forEach(repo => {
+          const name = repo.name;
+          createAndAppend('option', select, { text: name });
+        });
+
+        const repoInfo = createAndAppend('div', forRepoBlock);
+        const contribs = createAndAppend('div', forContributorsBlock);
+        select.addEventListener('change', evt => {
+          const selectedRepo = evt.target.value;
+          const repo = repositoryData.filter(r => r.name == selectedRepo)[0];
+          console.log(repo);
+          repoInfo.innerHTML = '';
+          contribs.innerHTML = '';
+
+          const addInfo = (label, value) => {
+            const container = createAndAppend('div', repoInfo);
+            createAndAppend('span', container, { text: label });
+            createAndAppend('span', container, { text: value });
+          };
+          addInfo('Name: ', repo.name);
+          addInfo('Desciption: ', repo.description);
+          addInfo('Number of forks: ', repo.forks);
+          addInfo('Updated: ', new Date(repo.updated_at));
+
+          const contribsUrl = repo.contributors_url;
+          
+          fetchJSON(contribsUrl)
+          .then((contribData) => {
+            contribData.forEach(contributor => {
+              // createAndAppend('p', contribs, { text: 'hej'});
+              createAndAppend('img', contribs, { src: contributor.avatar_url, height: 40, class: 'picture' });
```

Comment:
As in [week 1](https://github.com/mattiaslundberg/JavaScript3/pull/1/files#r316518610) I'm not a fan of this html structure as it doesn't provide any clue how to group the items. See linked comment for more details.